### DISCUSSION TITLE
about shadow map when camera moves near objects

### DISCUSSION THREAD

**Discussion by JackKaiXing** [google/filament#4083] [Q&A] [answered]
Hello,

Our project use just shadow map, no CSM. The Scene is composed of several different objects, which could be very tall or short.

When we move camera near the objects. if just the short objects is in the view space the shadow works file. However, when short and tall objects appears at the same time, the shadow works bad, sometime no shadow at all for small objects.

Even if only one tall object is in the view, when camera moves to the bottom of the object, its shadow not works fine. The border of the shadow spread out.

Any ideas or parameter could fix this?

Thanks.

**Comment by romainguy** [comment, marked as answer]
Do you have screenshot that show the problem?

**Reply by JackKaiXing** [reply]
![screenshot](https://user-images.githubusercontent.com/21079351/120751755-276b0700-c53b-11eb-97f8-0d931c072981.png)

**Reply by JackKaiXing** [reply]
we are using filement-1.9.18.

**Reply by JackKaiXing** [reply]
@romainguy any ideas？
